commit 0bc350dcfd034902b5ee9bafda28a1f28010d3c1 (refs/changes/85/5285/1)
Author: Mohamed Khalfella <mohamed.khalfella@joyent.com>
Date:   2018-12-22T00:03:04+00:00 (10 months ago)
    
    MANATEE-414 manatee pg_dump.sh should try to connect to all zookeeper instances
    before bailing out

diff --git a/pg_dump/pg_backup_common.sh b/pg_dump/pg_backup_common.sh
index ac95e14..577334c 100644
--- a/pg_dump/pg_backup_common.sh
+++ b/pg_dump/pg_backup_common.sh
@@ -25,7 +25,7 @@ DATE=
 DUMP_DATASET=
 DUMP_DIR=/var/tmp/upload/$(uuid)
 MANATEE_LOCK=/opt/smartdc/manatee/node_modules/.bin/manatee-adm
-MANATEE_STAT=manatee-stat
+MANATEE_ADM=/opt/smartdc/manatee/node_modules/manatee/bin/manatee-adm
 MANTA_DIR_PREFIX=/poseidon/stor/manatee_backups
 MMKDIR=mmkdir
 MPUT=mput
@@ -40,7 +40,7 @@ PG_START_TRIES=0
 UPLOAD_SNAPSHOT=
 ZFS_CFG=/opt/smartdc/manatee/etc/snapshotter.json
 ZFS_SNAPSHOT=$1
-ZK_IP=
+ZK_CS=
 
 function finish {
     if [[ $FATAL -ne 1 ]]; then
@@ -247,7 +247,7 @@ function get_self_role
     read -r shard_name_delim< <(echo $SHARD_NAME | gsed -e 's|\.|\\.|g')
 
     # figure out if we are the peer that should perform backups.
-    local shard_info=$($MANATEE_STAT $ZK_IP:2181 -s $SHARD_NAME)
+    local shard_info=$($MANATEE_ADM status -z $ZK_CS -s $SHARD_NAME)
     [[ -n $shard_info ]] || fatal "Unable to retrieve shardinfo from zookeeper"
 
     local async=$(echo $shard_info | json $shard_name_delim.async.ip)
diff --git a/pg_dump/pg_dump.sh b/pg_dump/pg_dump.sh
index 0f063ac..58d6232 100755
--- a/pg_dump/pg_dump.sh
+++ b/pg_dump/pg_dump.sh
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2018, Joyent, Inc.
 #
 
 # Postgres backup script. This script takes a snapshot of the current postgres
@@ -24,7 +24,7 @@ PG_DIR=
 UPLOAD_SNAPSHOT=
 MY_IP=
 SHARD_NAME=
-ZK_IP=
+ZK_CS=
 
 # mainline
 
@@ -35,6 +35,10 @@ if [[ -z "$1" ]]
         PG_START_TIMEOUT=$1
 fi
 
+BUILD_ZK_CS="this.zk_cs = this.zkCfg.servers.map(function (s) {"
+BUILD_ZK_CS+=" return (s.host + ':' + s.port);"
+BUILD_ZK_CS+="}).join(',');"
+
 DATASET=$(cat $ZFS_CFG | json dataset)
 [[ -n "$DATASET" ]] || fatal "unable to retrieve DATASET"
 DUMP_DATASET=zones/$(zonename)/data/pg_dump
@@ -44,8 +48,8 @@ MY_IP=$(mdata-get sdc:nics.0.ip)
 [[ -n "$MY_IP" ]] || fatal "Unable to retrieve our own IP address"
 # XXX get this dynamically somehow.
 SHARD_NAME=sdc
-ZK_IP=$(cat $CFG | json -a zkCfg.servers.0.host)
-[[ -n "$ZK_IP" ]] || fatal "Unable to retrieve nameservers from metadata"
+ZK_CS=$(cat $CFG | json -e "${BUILD_ZK_CS}" zk_cs)
+[[ -n "$ZK_CS" ]] || fatal "Unable to retrieve nameservers from metadata"
 
 get_self_role
 if [[ $? = '1' ]]; then
